### PR TITLE
Add more examples for metastore integration and glue

### PR DESCRIPTION
**Pull request by melodyyangaws**
added HMS sidecar, standalone helm chart, Hudi integration examples to the doc

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/metastore-integrations/docs/hive-metastore.md [hunk 1]
```diff
@@ -3,9 +3,12 @@
 
 ### **Hive metastore Database through JDBC**
```
Comment (dalbhanj) [inline content/metastore-integrations/docs/hive-metastore.md]: Looks like the content is the same as https://github.com/melodyyangaws/hive-emr-on-eks? I think solutions are great and gives customers flexibility to choose an architecture from 4 different options. I think being prescriptive would help choose the best architecture for their need. Can you summarize these use cases into separate paragraphs and mention why customers would choose one option over another, are there pros and cons of these approaches. For ex, I can think of using `start-job-run` API as a security risk because whoever has access to emr-containers api has access to sensitive info such as $USER_NAME, $PASSWORD etc. 
Once have these laid out, you should prioritize best architecture to least based on tradeoffs that are highlighted by your pros and cons

File: content/metastore-integrations/docs/hive-metastore.md [hunk 2]
```diff
@@ -151,4 +154,211 @@ aws emr-containers start-job-run --cli-input-json file:///Spark-Python-in-s3-hms
 
 ```
 
+###**Connect Hive metastore via thrift service hosted on EKS**
+In this example, our Spark application connects to a standalone Hive metastore service (HMS) running in EKS.
+
+Simply replace the environment varaibles in the following command, then install the [HMS helm chart](https://github.com/melodyyangaws/hive-emr-on-eks/tree/main/hive-metastore-chart) in EKS:
```
Comment (dalbhanj) [inline content/metastore-integrations/docs/hive-metastore.md]: Can we move the content to aws-samples repo rather than personal repo? https://github.com/melodyyangaws/hive-emr-on-eks/

**Review by dalbhanj — CHANGES_REQUESTED** [review]

File: content/metastore-integrations/docs/hive-metastore.md (hunk 2 above)
Comment (melodyyangaws) [reply content/metastore-integrations/docs/hive-metastore.md]: will move the repo to aws-sample and update the best practice doc later

File: content/metastore-integrations/docs/hive-metastore.md (hunk 1 above)
Comment (melodyyangaws) [reply content/metastore-integrations/docs/hive-metastore.md]: added pros and cons to the guide, see the additional commit below.

File: content/metastore-integrations/docs/hive-metastore.md [hunk 3]
```diff
@@ -151,4 +157,213 @@ aws emr-containers start-job-run --cli-input-json file:///Spark-Python-in-s3-hms
...
+      "entryPointArguments":["s3://'$S3BUCKET'"],
+      "sparkSubmitParameters": "--conf spark.driver.cores=1 --conf spark.executor.memory=4G --conf spark.driver.memory=1G --conf spark.executor.cores=2"}}' \
+--configuration-overrides '{
+    "monitoringConfiguration": {
+      "s3MonitoringConfiguration": {"logUri": "s3://'$S3BUCKET'/elasticmapreduce/emr-containers"}}}'
+```
+      
+### **4-Run thrift service as a sidecar in Spark Driver's pod**
+
+This advanced solution runs the standalone HMS thrift service inside a Spark driver as a sidecar. It means each Spark job will have its dedicated thrift server. The benefit of the design is HMS is no long a single point of failure, since each Spark application has its own HMS. Also it is no long a long running service, i.e. it spins up when your Spark job starts, then terminates when your job is done. The disadvantage is the initial setup against an existing EKS cluster is not easy and it requires combination skills of analytics and k8s.
```
Comment (dalbhanj) [inline content/metastore-integrations/docs/hive-metastore.md]: I'm not sure if I agree with disadvantage since you are already providing the solution. You may want to say maintenance of container image increases because you now have to maintain hms sidecar. Also you will need combination of analytics and k8s skill to maintain this solution

File: content/metastore-integrations/docs/hive-metastore.md [hunk 4]
```diff
@@ -151,4 +157,213 @@ aws emr-containers start-job-run --cli-input-json file:///Spark-Python-in-s3-hms
 
 ```
 
+### **3-Connect Hive metastore via thrift service hosted on EKS**
+In this example, our Spark application connects to a standalone Hive metastore service (HMS) running in EKS.
+
+Running the standalone HMS in EKS unifies your analytics applications with other business critical apps in a single platform. It simplifies your solution architecture and infrastructure design. The helm chart solution includes autoscaling feature, so your EKS cluster can automatically expand or shrink when the HMS request volume changes. Also it follows the security best practice to manage JDBC credentials via AWS Secrets Manager. However, the solution requires DevOps and k8s skills, also you need to be able to maintain the workload stability and resiliency in EKS.
```
Comment (dalbhanj) [inline content/metastore-integrations/docs/hive-metastore.md]: You can soften the tone a bit by saying this...
However, you will need combination of analytics and k8s skill to maintain this solution

File: content/metastore-integrations/docs/hive-metastore.md (hunk 3 above)
Comment (dalbhanj) [reply content/metastore-integrations/docs/hive-metastore.md]: Does the sidecar uses secrets from Secrets Manager? I don't remember if it did. If not, I'd add that as a con bcoz k8s secrets are base64 encoded and human readable. Customers may want to secure access to k8s secrets. Also, if they need secure solution, review your other solution which has integration with Secrets Manager

File: content/metastore-integrations/docs/hive-metastore.md [hunk 5]
```diff
@@ -151,4 +157,213 @@ aws emr-containers start-job-run --cli-input-json file:///Spark-Python-in-s3-hms
...
+          "spark.kubernetes.driver.podTemplateFile": "s3://'$S3BUCKET'/app_code/job/sidecar_hms_pod_template.yaml",
+          "spark.hive.metastore.uris": "thrift://localhost:9083"
+        }
+      }
+    ], 
+    "monitoringConfiguration": {
+      "s3MonitoringConfiguration": {"logUri": "s3://'$S3BUCKET'/elasticmapreduce/emr-containers"}}}'
+```
+
+### **5-Hudi + Remote Hive metastore integration**
```
Comment (dalbhanj) [inline content/metastore-integrations/docs/hive-metastore.md:282]: Is this a repeat of info from here?
content/metastore-integrations/docs/aws-glue.md

**Review by dalbhanj — CHANGES_REQUESTED** [review]
Thanks for adding additional info. Few minor changes requested and we should be good

File: content/metastore-integrations/docs/hive-metastore.md (hunk 5 above)
Comment (melodyyangaws) [reply content/metastore-integrations/docs/hive-metastore.md:282]: no, this is Hudi sync with remote RDS hive metastore, the other one is sync with Glue catalog.

**Review by dalbhanj — APPROVED** [review]
LGTM
